diff --git a/win32/makefile.w32 b/win32/makefile.w32
index 0608146..df670f7 100644
--- a/win32/makefile.w32
+++ b/win32/makefile.w32
@@ -17,6 +17,15 @@ NODEBUG=1
 
 # Uncomment the following macro to use the optimized assembler
 # routines in Zip:
+!IFDEF X64ASM
+AS=ml64 -nologo
+ASFLAGS= /c /Zi /DINFOZIP
+ASMOBJS = gvmat64.obj
+CRCA_O =
+CFLG_ASM = -DNO_ASM_CRC -DASMV
+!ELSE
+AS=ml -nologo
+ASFLAGS=-c -coff -Cx
 !IFDEF NOASM
 ASMOBJS =
 CRCA_O =
@@ -26,6 +35,7 @@ ASMOBJS = match32.obj
 CRCA_O = crci386c.obj
 CFLG_ASM = -DASM_CRC
 !ENDIF
+!ENDIF
 
 !IFDEF USEBZ2
 LOC=$(LOC) -DBZIP2_SUPPORT
@@ -80,11 +90,6 @@ CC=cl -nologo
 CFLAGS=-W3 $(cdebug) -DWIN32 $(CFLG_ASM) $(CRTLIB) $(LOC)
 UTILFLAGS=$(CFLAGS) -DUTIL -Fo$@
 
-# Remove "-coff" from ASFLAGS if you do not have MASM 6.11.
-
-AS=ml -nologo
-ASFLAGS=-c -coff -Cx
-
 RC=rc
 
 # If you build 16-bit executables with MS Visual C++ v1.0/1.5 and link them
@@ -204,6 +209,9 @@ crc_i386.obj:   win32/crc_i386.asm
 match32.obj:    win32/match32.asm
  $(AS) $(ASFLAGS) win32\match32.asm
 
+gvmat64.obj:    win32/gvmat64.asm
+    $(AS) $(ASFLAGS) win32\gvmat64.asm
+
 zip.res:        win32/zip.rc revision.h
 	$(RC) /l 0x409 /fo$@ /i win32 /d WIN32 win32/zip.rc
 
